web/src/lib/components/assets/thumbnail/thumbnail.svelte

```diff
@@ -344,17 +351,31 @@
         </div>
       {/if}
 
-      {#if (!loaded || thumbError) && asset.thumbhash}
-        <canvas
-          use:thumbhash={{ base64ThumbHash: asset.thumbhash }}
-          data-testid="thumbhash"
-          class="absolute top-0 object-cover"
-          style:width="{width}px"
-          style:height="{height}px"
-          class:rounded-xl={selected}
-          draggable="false"
-          out:fade={{ duration: THUMBHASH_FADE_DURATION }}
-        ></canvas>
+      {#if asset.thumbhash}
+        {#await isCached(new Request(thumbnailURL))}
+          <canvas
```

Comment:
I added a second canvas which doesn't fade out, it's used whilst it's waiting to check if the thumbnail is already in the cache. If the thumbnail is cached, then it is removed immediately, whereas if the thumbnail is not cached then the other canvas is used and that will fade out when the actual thumbnail is ready to use.